module/data/actor/templates/common.mjs

```diff
@@ -0,0 +1,67 @@
+import SystemDataModel from "../../abstract.mjs";
+import { MappingField } from "../../fields.mjs";
+import CurrencyTemplate from "../../shared/currency.mjs";
+import AbilityData from "../ability.mjs";
+
+/**
+ * A template for all actors that share the common template.
+ *
+ * @property {Object<string, AbilityData>} abilities  Actor's abilities.
+ * @mixin
+ */
+export default class CommonTemplate extends SystemDataModel.mixin(CurrencyTemplate) {
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return this.mergeSchema(super.defineSchema(), {
+      abilities: new MappingField(new foundry.data.fields.EmbeddedDataField(AbilityData), {
+        initialKeys: CONFIG.DND5E.abilities
+      }, {label: "DND5E.Abilities"})
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static migrateData(source) {
+    this.migrateACData(source);
+    this.migrateMovementData(source);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Migrate the actor ac.value to new ac.flat override field.
+   * @param {object} source  The candidate source data from which the model will be constructed.
+   */
+  static migrateACData(source) {
+    if ( !source.attributes?.ac ) return;
+    const ac = source.attributes.ac;
+
+    // If the actor has a numeric ac.value, then their AC has not been migrated to the auto-calculation schema yet.
+    if ( Number.isNumeric(ac.value) ) {
+      ac.flat = parseInt(ac.value);
+      ac.calc = this._systemType === "npc" ? "natural" : "flat";
```

Comment:
Can we avoid the complexity of needing a `static _systemType` by just making an assumption here like picking `"flat"` if we have to do a migration from a numeric value?

Reply:
I remember we took some pains to make sure NPC actors were given a `natural` AC as part of this migration. We could move it back to the general `migration.mjs` code. The `_systemType` concept might come in useful elsewhere though.